packages/mui-base/src/SnackbarUnstyled/SnackbarUnstyled.tsx

```diff
@@ -0,0 +1,337 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { OverridableComponent } from '@mui/types';
+import { unstable_useEventCallback as useEventCallback } from '@mui/utils';
+import ClickAwayListener from '../ClickAwayListener';
+import {
+  SnackbarCloseReason,
+  SnackbarUnstyledProps,
+  SnackbarUnstyledTypeMap,
+} from './SnackbarUnstyled.types';
+import composeClasses from '../composeClasses';
+import { getSnackbarUnstyledUtilityClass } from './snackbarUnstyledClasses';
+
+const useUtilityClasses = () => {
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getSnackbarUnstyledUtilityClass, undefined);
+};
+/**
+ *
+ * Demos:
+ *
+ * - [Snackbar](https://mui.com/base/react-snackbar/)
+ *
+ * API:
+ *
+ * - [SnackbarUnstyled API](https://mui.com/base/api/snackbar-unstyled/)
+ */
+const SnackbarUnstyled = React.forwardRef(function SnackbarUnstyled(
+  props: SnackbarUnstyledProps & { component?: React.ElementType },
+  ref: React.ForwardedRef<any>,
+) {
+  const {
+    autoHideDuration = null,
+    children,
+    className,
+    ClickAwayListenerProps,
+    component,
+    components = {},
+    componentsProps = {},
+    disableWindowBlurListener = false,
+    onBlur,
+    onClose,
+    onFocus,
+    onMouseEnter,
+    onMouseLeave,
+    open,
+    resumeHideDuration,
+    ...other
+  } = props;
+
+  const classes = useUtilityClasses();
+
+  const timerAutoHide = React.useRef<ReturnType<typeof setTimeout>>();
```

Comment:
Would be great if the logic is extracted in a hook `useSnackbar` so that we can reuse it in Joy UI and Material UI afterwards.

Reply:
Done.